sourced/ml/__main__.py

```diff
@@ -33,6 +35,21 @@ def add_parser(name, help_message):
             name, help=help_message, formatter_class=ArgumentDefaultsHelpFormatterNoNone)
 
     # ------------------------------------------------------------------------
+    preprocessing_parser = subparsers.add_parser(
+        "preprocrepos", help="Convert siva to parquet files with extracted information.")
+    preprocessing_parser.set_defaults(handler=preprocess_repos_entry)
+    preprocessing_parser.add_argument("-x", "--mode", choices=Moder.Options.__all__,
+                                      default="file", help="What to extract from repositories.")
+    add_repo2_args(preprocessing_parser)
+    add_dzhigurda_arg(preprocessing_parser)
+    preprocessing_parser.add_argument(
+        "-o", "--output", required=True,
+        help="[OUT] Path to the parquet files with bag batches.")
+    default_fields = ["blob_id", "repository_id", "content", "path", "commit_hash", "uast"]
+    preprocessing_parser.add_argument(
+        "-f", "--fields", action="append", default=default_fields,
```

Comment:
`choices=default_fields` it is not so good, because we can ask for any field from the Siva file. They may be changed, so I do not to have any constrains here. Let keep it as 
```python
parser.add_argument("-f", "--fields", nargs="+", default=default_fields,
        help="Fields to save.")
```

Reply:
ok if you think its best